src/Kiota.Builder/Writers/TypeScript/CodeFunctionWriter.cs

```diff
@@ -178,79 +365,126 @@ private void WritePropertySerializer(string modelParamName, CodeProperty codePro
             {
                 writer.WriteLine($"if({modelParamName}.{codePropertyName})");
             }
-            writer.WriteLine($"writer.{serializationName}(\"{codeProperty.WireName}\", {spreadOperator}{modelParamName}.{codePropertyName}{defaultValueSuffix});");
+            if (composedType is not null && IsComposedOfPrimitives(composedType))
+                writer.WriteLine($"{serializationName}(writer, \"{codeProperty.WireName}\", {spreadOperator}{modelParamName}.{codePropertyName}{defaultValueSuffix});");
+            else
+                writer.WriteLine($"writer.{serializationName}(\"{codeProperty.WireName}\", {spreadOperator}{modelParamName}.{codePropertyName}{defaultValueSuffix});");
         }
     }
 
-    private string GetSerializationMethodName(CodeTypeBase propType)
+
+    public string GetSerializationMethodName(CodeTypeBase propertyType, CodeMethod method)
     {
-        var propertyType = conventions.TranslateType(propType);
-        if (!string.IsNullOrEmpty(propertyType) && propType is CodeType currentType && GetSerializationMethodNameForCodeType(currentType, propertyType) is string result && !string.IsNullOrWhiteSpace(result))
+        ArgumentNullException.ThrowIfNull(propertyType);
+        ArgumentNullException.ThrowIfNull(method);
+
+        var composedType = GetOriginalComposedType(propertyType);
+        if (composedType is not null && IsComposedOfPrimitives(composedType))
```

Comment:
don't we already have some of that at least with GetFunctionName?

Reply:
I meant the override with just two parameters

Reply:
Alright, I'll clean it up